### DISCUSSION TITLE
#[async_trait] on SessionStore and ExpiredDeletion traits

### DISCUSSION THREAD

**Discussion by mispp** [maxcountryman/tower-sessions#219] [Q&A] [answered]
Beginner here,

is there a reason why `async_trait` `proc_macro` is still used given that since 2023 `async fn` [are allowed in trait objects](https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits.html)?
Would this be possible to easily update?

Came to this while analyzing strongly-typed example.

**Comment by maxcountryman** [comment, marked as answer]
It's needed for now due to object safety.

I haven't dug very deep, so I'm unclear if that's a limitation that will likely always exist or not.

**Reply by mispp** [reply]
Not sure if [this](https://blog.rust-lang.org/2023/12/21/async-fn-rpit-in-traits.html#should-i-still-use-the-async_trait-macro) helps.

Basically it says there are two reasons why to still use `async_trait` macro:
- You want to support Rust versions older than 1.75.
- You want dynamic dispatch.

Also

> As stated above, we hope to enable dynamic dispatch in a future version of the trait-variant crate.

Since im a beginner, this doesnt mean much to me.

**Reply by maxcountryman** [reply]
Relative to this crate, it means `async_trait` is still required.
